tools/cmake/app.cmake

```diff
@@ -23,3 +23,14 @@ if(NOT CMAKE_TOOLCHAIN_FILE OR MBED_TOOLCHAIN_FILE_USED)
 endif()
 
 enable_language(C CXX ASM)
+
+# set executable suffix (has to be done after enabling languages)
+# Note: This is nice to have, but is also required because STM32Cube will only work on files with a .elf extension
+set(CMAKE_EXECUTABLE_SUFFIX .elf)
```

Comment:
I recall other IDE also had this limitation (only able to find executable in their viewer if it has specific extension, it could have been `.axf` for uvision). However it should work (this shall be tested before we hit the problem like STM32Cube in this case).